rust/ql/lib/codeql/rust/Concepts.qll

```diff
@@ -264,3 +266,45 @@ module Cryptography {
 
   class CryptographicAlgorithm = SC::CryptographicAlgorithm;
 }
+
+/** Provides classes for modeling path-related APIs. */
+module Path {
+  /**
+   * A data-flow node that performs path normalization. This is often needed in order
+   * to safely access paths.
+   */
+  class PathNormalization extends DataFlow::Node instanceof PathNormalization::Range {
+    /** Gets an argument to this path normalization that is interpreted as a path. */
+    DataFlow::Node getPathArg() { result = super.getPathArg() }
+  }
+
+  /** Provides a class for modeling new path normalization APIs. */
+  module PathNormalization {
+    /**
+     * A data-flow node that performs path normalization. This is often needed in order
+     * to safely access paths.
+     */
+    abstract class Range extends DataFlow::Node {
+      /** Gets an argument to this path normalization that is interpreted as a path. */
+      abstract DataFlow::Node getPathArg();
+    }
+  }
```

Comment:
`PathNormalization` and `PathNormalization::Range` appear to have no implementations, now that we have the `normalize-path` optional step and MaD model.  Is there a reason to keep these classes?

Reply:
They are meant to be extension points provided by the Concepts library.

Reply:
OK, I'm happy enough we keep them as an extension point.